Classes/EventStore/Storage/Doctrine/DataTypes/DateTimeType.php

```diff
@@ -0,0 +1,80 @@
+<?php
+namespace Neos\Cqrs\EventStore\Storage\Doctrine\DataTypes;
+
+/*
+ * This file is part of the Neos.EventStore.DatabaseStorageAdapter package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * DateTimeType
+ */
+class DateTimeType extends Type
+{
+    const DATETIME_MICRO = 'datetime_micro';
+    const MICRO_FORMAT = 'Y-m-d H:i:s.u';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return self::DATETIME_MICRO;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform)
+    {
+        $declaration = 'DATETIME';
+        $declaration .= ' COMMENT \'(DC2Type:' . self::DATETIME_MICRO . ')\'';
```

Comment:
This is currently hardcoded to MySQL and won't work with older versions.
For the sake of compatibility & simplicity I'd suggest to use a regular DATETIME column for now (if one needs the ms they can still be added to the metadata/payload of events).. Anyways, separate topic